lib/src/ui/watch/tv_screen.dart

```diff
@@ -26,18 +35,71 @@ class _TvScreenState extends ConsumerState<TvScreen>
     with RouteAware, WidgetsBindingObserver {
   @override
   Widget build(BuildContext context) {
-    return PlatformWidget(
+    return ConsumerPlatformWidget(
+      ref: ref,
       androidBuilder: _androidBuilder,
       iosBuilder: _iosBuilder,
     );
   }
 
+  void showChoicesDialog(WidgetRef ref) {
+    ChannelVariant selectedVariant = ref.watch(variantStateProvider);
+
+    final repo = ref.watch(tvRepositoryProvider);
+    final FutureResult<TvChannels> channelResult = repo.getTvChannels();
+
+    channelResult.then((res) {
+      if (res.isValue) {
+        final TvChannels data = res.asValue!.value;
+        showChoicePicker(
+          context,
+          choices: ChannelVariant.values,
+          selectedItem: selectedVariant,
+          labelBuilder: (t) => Text(t.title),
+          onSelectedItemChanged: (ChannelVariant? d) {
+            if (d != null) {
+              selectedVariant = d;
+            }
+          },
+        ).then((_) {
+          String? newGameId = data.channels.entries
+              .where((element) => element.key == selectedVariant.title)
+              .first
+              .value
+              .gameId;
+          if (selectedVariant.title == "Top Rated") newGameId = null;
+
+          ref.read(variantStateProvider.notifier).state = selectedVariant;
+          ref
+              .read(_featuredGameWithSoundProvider.notifier)
+              .setGameId(newGameId);
+        });
+      }
+    }); // End FR_data.then
+  }
+
   Widget _androidBuilder(
     BuildContext context,
+    WidgetRef ref,
   ) {
+    final ChannelVariant selectedVariant = ref.watch(variantStateProvider);
     return Scaffold(
       appBar: AppBar(
-        title: const Text('Top Rated'),
+        title: InkWell(
```

Comment:
You should not use `InkWell` but rather `AppBarTextButton` from `widgets/buttons.dart`